src/software/ai/hl/stp/tactic/receiver/receiver_fsm.h

```diff
@@ -141,23 +158,37 @@ struct ReceiverFSM
      */
     bool strayPass(const Update& event);
 
+    /**
+     * Guard that checks if the pass went astray or if the ball is slowing down.
+     *
+     * @param event ReceiverFSM::Update event
+     *
+     * @return true if the pass went astray or if the ball is slowing down
+     */
+    bool strayOrSlowPass(const Update& event);
+
     auto operator()()
     {
         using namespace boost::sml;
 
+        DEFINE_SML_STATE(WaitingForPassState)
         DEFINE_SML_STATE(ReceiveAndDribbleState)
         DEFINE_SML_STATE(OneTouchShotState)
-        DEFINE_SML_STATE(WaitingForPassState)
+        DEFINE_SML_STATE(DribbleSkillFSM)
+
         DEFINE_SML_EVENT(Update)
 
         DEFINE_SML_GUARD(onetouchPossible)
         DEFINE_SML_GUARD(passStarted)
-        DEFINE_SML_GUARD(passFinished)
+        DEFINE_SML_GUARD(passReceived)
+        DEFINE_SML_GUARD(passReceivedByTeammate)
         DEFINE_SML_GUARD(strayPass)
+        DEFINE_SML_GUARD(strayOrSlowPass)
```

Comment:
nit: since `strayPass` already exists as a guard, could a `slowPass` guard and just using boolean logic in the FSM transition diagram be just as clear?